crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -208,10 +286,18 @@ impl ServerBuilder {
         Ok(cors)
     }
 
+    /// Consumes the `ServerBuilder` to create a `Server` that can be run.
     pub fn build(self) -> Result<Server, Error> {
-        let (address, schema, router) = self.build_components();
+        let state = self.state.clone();
+        let (address, schema, db_reader, router) = self.build_components();
+
+        // Start the background task to update a shared `CheckpointViewedAt`
+        let watermark = Arc::new(AtomicU64::new(0));
+        let watermark_clone = watermark.clone();
+
         let app = router
             .layer(axum::extract::Extension(schema))
+            .layer(axum::extract::Extension(watermark))
```

Comment:
Something I'm learning about myself: I really don't like the `_clone` pattern...

```suggestion
        // Start the background task to update a shared `CheckpointViewedAt`
        let checkpoint_watermark = Arc::new(AtomicU64::new(0));

        let app = router
            .layer(axum::extract::Extension(schema))
            .layer(axum::extract::Extension(checkpoint_watermark.clone()))
```